### PR TITLE
Big rename

### PR DESCRIPTION
**Pull request by uber-qlam**
- Create a 'persistence-tests' folder for all the tests except for the history serializer tests
-- this prevents a package import cycle between 'persistence' and 'cassandra' (say)
- Put cassandra and sql implementations in their separate places
- Add some files that contain global constants required by both impls
-- To make the reorg possible, some methods e.g. `deserializeClusterConfigs` also had to become public (e.g. in this case, moved to `dataInterfaces.go`). Whether this is possibly the right place to put all common constants/methods is not certain.

### CODE REVIEW

File: common/persistence/sql/sqlMetadataPersistence.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 	"fmt"
 
 	workflow "github.com/uber/cadence/.gen/go/shared"
+	persistence "github.com/uber/cadence/common/persistence"
```
Comment (meiliang86) [inline common/persistence/sql/sqlMetadataPersistence.go]: Redundant alias?

Comment (meiliang86) [reply common/persistence/sql/sqlMetadataPersistence.go]: Also, can you rename FlatCommon to DomainCommon?

File: common/persistence/sql/sqlMetadataPersistence.go [hunk 2]
```diff
@@ -154,14 +155,14 @@ WHERE
 name = :name AND
 id = :id`
 
-	templateDeleteDomainByIdSqlQuery   = `DELETE FROM domains WHERE id = :id`
-	templateDeleteDomainByNameSqlQuery = `DELETE FROM domains WHERE name = :name`
+	templateDeleteDomainByIDSQLQuery   = `DELETE FROM domains WHERE id = :id`
```
Comment (meiliang86) [inline common/persistence/sql/sqlMetadataPersistence.go]: Let's remove the "template" in the beginning and just call it XXXSQLQuery

File: common/persistence/persistenceCommon.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package persistence
+
+const (
+	TransferTaskTransferTargetWorkflowID = "20000000-0000-f000-f000-000000000001"
```
Comment (meiliang86) [inline common/persistence/persistenceCommon.go]: This is only referenced in test file. Where in the code do we actually use it?

Comment (meiliang86) [reply common/persistence/persistenceCommon.go]: Also, we can put these in dataInterfaces.go for now. I don't think we need two new files for constants.

File: common/persistence/sql/sqlMetadataPersistence.go (hunk 1 above)
Comment (meiliang86) [reply common/persistence/sql/sqlMetadataPersistence.go]: domainDataRow does not seem to be used/referenced. Remove?

File: service/frontend/service.go [hunk 4]
```diff
@@ -102,7 +103,7 @@ func (s *Service) Start() {
 	}
 	visibility = persistence.NewVisibilityPersistenceClient(visibility, base.GetMetricsClient(), log)
 
-	history, err := persistence.NewCassandraHistoryPersistence(p.CassandraConfig.Hosts,
+	history, err := cassandra.NewCassandraHistoryPersistence(p.CassandraConfig.Hosts,
```
Comment (meiliang86) [inline service/frontend/service.go]: Remove the Cassandra part in method name now as it's in cassandra package. We can change it to  cassandra.NewHistoryPersistence. Same for others.

**Review by meiliang86 — APPROVED** [review]
LGTM overall. Some minor comments about namings. You can merge it after you address all the comments. I don't need to take a second look at this.

File: common/persistence/persistenceCommon.go (hunk 3 above)
Comment (uber-qlam) [reply common/persistence/persistenceCommon.go]: forgot to rename the constant in `cassandraPersistence.go`

**Pull request merged**